models/demos/falcon7b/tt/falcon_mlp.py

```diff
@@ -288,17 +289,19 @@ def _load_mlp_padded_tensors(self):
         self.model_config["MLP_DECODE_PADDING_TENSORS"] = tt_paddings
 
     def forward(self, x: tt_lib.tensor.Tensor) -> tt_lib.tensor.Tensor:
+        batch_size = x[0].shape[-2]  # assume all devices have same shape
         hidden_states = []
-        for device_id in range(len(x)):
-            # pad inputs with padding tensor if not already padded
-            if (
-                self.model_config["PREFILL_OPTIMIZED_MODE"]
-                and x[device_id].shape[-1] < self.padding_value
-                and self.prefill_seq_len in [1024, 2048]
-            ):
-                x[device_id] = ttnn.concat(
+        # pad inputs with padding tensor if not already padded
+        if (
+            self.model_config["PREFILL_OPTIMIZED_MODE"]
+            and self.hidden_size < self.padding_value
+            and self.prefill_seq_len in [1024, 2048]
+        ):
+            for device_id in range(self.num_devices):
+                x[device_id] = tt_lib.tensor.concat(
```

Comment:
Can you call ttnn.experimental instead? And the same for unpad op below?

Reply:
Yes, all ttlib ops will be converted to tnn experimental in an upcoming PR, as I mentioned in the comment above.